### ISSUE TITLE
functions with multiple address ranges

### ISSUE THREAD

**Issue by wideglide** [gimli-rs/ddbug#32]
Do you plan to change the function implementation to support functions with multiple ranges?  It seems this is supported in [addr2line](https://github.com/gimli-rs/addr2line/blob/e1f7b4d2b2aa3da58d63793c2caa5d2ed4e0c08c/src/function.rs#L24), but here they are presumed to have only one range.

**Comment by philipc** [comment]
Yes, definitely need to support that.

**Comment by philipc** [comment]
Do you have an easy way to generate such a function for testing?

**Comment by wideglide** [comment]
This is an example, but outside of discovering samples, I don't know how to generate one.

[busybox-1_21_stable_gcc-11.2.0_x86_64_O3_busybox_unstripped.gz](https://github.com/gimli-rs/ddbug/files/8211799/busybox-1_21_stable_gcc-11.2.0_x86_64_O3_busybox_unstripped.gz)

There's a few in this sample, they all have a second range labeled `<function_name>.cold` in the symbol table.
Here's the dwarfdump of `build_tree`:
```
0x000faff3:   DW_TAG_subprogram
                DW_AT_name      ("build_tree")
                DW_AT_decl_file ("/data/ase18/sources/busybox-1_21_stable_gcc-11.2.0_x86_64_O3_debug/archival/gzip.c")
                DW_AT_decl_line (1223)
                DW_AT_decl_column       (0x0d)
                DW_AT_prototyped        (true)
                DW_AT_ranges    (0x00010944
                   [0x00000000004a1950, 0x00000000004a23e2)
                   [0x0000000000407a30, 0x0000000000407b0c))
                DW_AT_frame_base        (DW_OP_call_frame_cfa)
                DW_AT_call_all_tail_calls       (true)
```

**Comment by philipc** [comment]
Fixed in a396d23d0bf6a28be0ec924193fa4270c628d11c
